app/models/step.rb

```diff
@@ -12,18 +12,18 @@ class Approval < ActiveRecord::Base
   acts_as_list scope: :proposal
   validates :proposal, presence: true
 
-  belongs_to :parent, class_name: 'Approval'
-  has_many :child_approvals, class_name: 'Approval', foreign_key: 'parent_id', dependent: :destroy
-
-  scope :individual, -> { where(type: 'Approvals::Individual') }
+  belongs_to :parent, class_name: "Step"
+  has_many :child_approvals, class_name: "Step", foreign_key: "parent_id", dependent: :destroy
 
+  # @TODO: Auto-generate list of subclasses
+  scope :individual, -> { where(type: ["Steps::Approval", "Steps::Purchase"]).order("position ASC") }
```

Comment:
I would add `order` as a separate scope. Could be overlooked when it is part of this scope. Either that, or rename to be something like `individial_in_position_order`